lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2055,9 +2055,42 @@ static NodeValue convertConstant(Module &mod, Constant &constant,
   }
 }
 
+/// Compute number of significant bits that are used to represent data of type
+/// \p kind. For FP, it is the number of bits in mantissa, for integers it's the
+/// number of bits except sign bit.
+/// \p returns number of significant bits of \p kind.
+static size_t numSignificantBits(ElemKind kind) {
+  switch (kind) {
+  case ElemKind::Int8QTy:
+    return std::numeric_limits<int8_t>::digits;
+  case ElemKind::UInt8QTy:
+  case ElemKind::UInt8FusedQTy:
+    return std::numeric_limits<uint8_t>::digits;
+  case ElemKind::Float16Ty:
+    // Custom type with layout 0 00000 0000000000.
+    return 10;
+  case ElemKind::Int16QTy:
+    return std::numeric_limits<int16_t>::digits;
+  case ElemKind::FloatTy:
+    return std::numeric_limits<float>::digits;
+  case ElemKind::Int32QTy:
+  case ElemKind::Int32ITy:
+    return std::numeric_limits<int32_t>::digits;
+  case ElemKind::Int64ITy:
+    return std::numeric_limits<int64_t>::digits;
+
+  default:
+    llvm_unreachable("Unknown type");
+  }
+  return 0;
```

Comment:
nit: Don't think we will need to return here, since the switch will cover all possible cases.

Reply:
We still need to have something here (at least llvm_unreachable), otherwise the compiler complains that control reaches the end of non-void function.

Reply:
Hmm ok, I think we do this in many other places but it's NBD.